### PR TITLE
Daniukoi khydzik lvneta 89 adding registration authorization

### PR DESCRIPTION
**Pull request by daniukoi**
Added authorisation and registration

### CODE REVIEW

File: TheaterSchedule.BLL/Classes/Form.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace TheaterSchedule.BLL.Classes
+{
+    public class Form
+    {
+        public List<Multiple_Form> Multiple;
+        public List<Single_Form> Single;
+        public List<Radio_Form> Radio;
+
+        public Form()
```
Comment (vtorkalo) [inline TheaterSchedule.BLL/Classes/Form.cs]: looks like this is not related to current PR

File: TheaterDatabase/dbo/Tables/Performance_TR.sql [hunk 2]
```diff
@@ -1,11 +0,0 @@
-﻿CREATE TABLE [dbo].[Performance_TR]
-(
-	[Performance_TRId] INT IDENTITY(1,1) NOT NULL,
-	[Title] NVARCHAR(50) NOT NULL,
-	[LanguageId] INT NOT NULL,
```
Comment (vtorkalo) [inline TheaterDatabase/dbo/Tables/Performance_TR.sql]: also not related to current PR. please revert not needed changes

File: TheaterSchedule.BLL/Services/RefreshTokenService.cs [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                throw new HttpStatusCodeException(
+                       HttpStatusCode.NotFound, $"Such refresh token doesn't exist");
+            }
+
+            refreshToken.RefreshToken = refreshtoken;
+            refreshToken.DaysToExpire = DateTime.Now.AddDays(daysToExpire);
+
+            await _refreshTokenRepository.UpdateAsync(refreshToken);
+          
+            await _theaterScheduleUnitOfWork.SaveAsync();
```
Comment (vtorkalo) [inline TheaterSchedule.BLL/Services/RefreshTokenService.cs]: use UnitOfWork with using() statement to call IDisposable

File: TheaterSchedule.BLL/Services/TokenService.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        public string GenerateAccessToken(ApplicationUserDTO user)
+        {
+            var claims = new List<Claim>
+            {
+                new Claim("userId", user.Id.ToString()),
+                new Claim("firstName", user.FirstName),
+                new Claim("lastName", user.LastName),
+                new Claim("email", user.Email),
+                new Claim("pnoneNumber", user.PnoneNumber),
+                new Claim("dateOfBirth", user.DateOfBirth.ToString())
```
Comment (vtorkalo) [inline TheaterSchedule.BLL/Services/TokenService.cs]: extract strings to static class with constants
Like ClaimKeys

File: TheaterSchedule.DAL/Interfaces/IRefreshTokenRepository.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+using System;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace TheaterSchedule.DAL.Interfaces
+{
+    public interface IRefreshTokenRepository
+    {
+        RefreshTokens Insert(RefreshTokens entity);
+        Task<RefreshTokens> InsertAsync(RefreshTokens entity);
```
Comment (vtorkalo) [inline TheaterSchedule.DAL/Interfaces/IRefreshTokenRepository.cs]: rename entity to token - for better readability

File: TheaterSchedule/Controllers/RegistrationController.cs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+
+            try
+            {
+                _service.Create(userDTO, userDTO.Password);
+                return StatusCode(201, userDTO);
+            }
+            catch (Exception e)
```
Comment (vtorkalo) [inline TheaterSchedule/Controllers/RegistrationController.cs]: don't need catch - we already have exception handling middleware

Comment (daniukoi) [reply TheaterSchedule/Controllers/RegistrationController.cs]: Fix this.
